test/zfcp_controller_test.rb

```diff
@@ -5,22 +5,96 @@
 Yast.import "ZFCPController"
 
 describe "Yast::ZFCPController" do
+  before do
+    Yast::ZFCPController.main
+  end
 
   describe "#ActivateDisk" do
-    it "Activates the given disk" do
-      expect(Yast::SCR).to receive(:Execute).with(anything, /\/sbin\/zfcp_host_configure '1' 1/).and_return(0)
-      expect(Yast::ZFCPController).to_not receive(:ReportControllerActivationError)
-      Yast::ZFCPController.ActivateDisk(1, "", "")
+    before do
+      allow(Yast::Arch).to receive(:is_zkvm).and_return(false)
+      allow(Yast::SCR).to receive(:Read).with(Yast.path(".probe.storage")).once
+        .and_return(load_data("probe_storage.yml"))
+      allow(Yast::SCR).to receive(:Read).with(Yast.path(".probe.disk")).once
+        .and_return(load_data("probe_disk.yml"))
+      allow(Yast::SCR).to receive(:Read).with(Yast.path(".probe.tape")).once.and_return([])
+    end
+
+    it "activates the controller and the given disk" do
+      expect(Yast::ZFCPController).to receive(:activate_controller).with("0.0.fa00")
+      expect(Yast::SCR).to receive(:Execute)
+        .with(anything, /\/sbin\/zfcp_disk_configure '0.0.fa00' '0x500\d+' '0x401\d+' 1/)
+        .and_return(0)
+      Yast::ZFCPController.ActivateDisk("0.0.fa00", "0x5000000000000000", "0x4010400000000000")
+    end
+
+    context "when the disk is already active" do
+      it "does not try to active the given disk" do
+        allow(Yast::ZFCPController).to receive(:activate_controller).with("0.0.fa00")
+        expect(Yast::SCR).to_not receive(:Execute)
+        Yast::ZFCPController.ActivateDisk("0.0.fa00", "0x500507630500873a", "0x4010400000000000")
```

Comment:
any reason to not use subject here ( and here I mean whole file )? which is shorter :)

Reply:
No, just for consistency with the rest of the file :) But you are right, I have changed it.